ordered_set.py

```diff
@@ -49,9 +55,9 @@ class OrderedSet(MutableSet, Sequence):
         OrderedSet([1, 2, 3])
     """
 
-    def __init__(self, iterable=None):
-        self.items = []
-        self.map = {}
+    def __init__(self, iterable: Union[Iterable[T], Set[T]] = None):
```

Comment:
A `Set[T]` is an `Iterable[T]`, so you can simplify this to `Iterable[T]`. 

Also, best practice would be to mark this `Optional[Iterable[T]] = None` rather than using an "implicit optional".